service/src/test/java/uk/gov/hmcts/reform/fpl/utils/ConfidentialDetailsHelperTest.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.hmcts.reform.fpl.utils;
+
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.reform.fpl.model.Address;
+import uk.gov.hmcts.reform.fpl.model.Other;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+
+import java.util.List;
+import java.util.UUID;
+
+import static java.util.UUID.randomUUID;
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.fpl.utils.ConfidentialDetailsHelper.addConfidentialDetailsForElement;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.element;
+
+class ConfidentialDetailsHelperTest {
+
+    private static final String CONFIDENTIAL = "Yes";
+    private static final UUID ID = randomUUID();
+    private static final String TELEPHONE_NUMBER = "01227 831393";
+
+    @Test
+    void shouldAddConfidentialDetailsForOther() {
+        List<Element<Other>> others = List.of(otherWithConfidentialFields(ID, CONFIDENTIAL));
+        Element<Other> othersNotConfidential = otherWithRemovedConfidentialFields(ID);
+
+        Other confidentialOthers = addConfidentialDetailsForElement(others, othersNotConfidential);
+
+        assertThat(confidentialOthers.getTelephone()).isEqualTo(TELEPHONE_NUMBER);
+    }
+
+    @Test
+    void shouldNotAddConfidentialDetailsForNonMatchingElement() {
+        List<Element<Other>> others = List.of(otherWithConfidentialFields(ID, CONFIDENTIAL));
+        Element<Other> othersNotConfidential = otherWithRemovedConfidentialFields(randomUUID());
+
+        Other confidentialOthers = addConfidentialDetailsForElement(others, othersNotConfidential);
+
+        assertThat(confidentialOthers).isEqualTo(otherWithRemovedConfidentialFields(ID).getValue());
```

Comment:
Again it's not adding any confidential details here, if it doesn't find a match in confidential it returns the value of the element you passed in